sympy/matrices/eigen.py

```diff
@@ -573,7 +573,7 @@ def _is_negative_semidefinite(M):
 
 def _is_indefinite(M):
     if M.is_hermitian:
-        eigen        = M.eigenvals()
+        eigen = M.eigenvals()

```

Comment:
Unneeded change

Reply:
As for the `as_immutable` yes it is a hack and I don't like it either for the same reason you don't, but since the `_MinimalMatrix` seems to exist only for testing I thought it is the easiest way to get the tests working correctly with the caching changes in #18468 which is what it is really meant for. Without this method the tests which test individual classes via `class ArithmeticOnlyMatrix(_MinimalMatrix, MatrixArithmetic):` will fail. The other option is to change all those tests to simply use `Matrix`. Or have a look at that PR and if you have another suggestion I will certainly listen. The spacing I can change back wherever it bothers you but I will make that change with the push with whatever solution you propose for the `as_immutable` test cases.

Reply:
The description of `_MinimalMatrix` says that it serves as a reference for what methods are required of all matrices. If we add `as_immutable()`, that's no longer true. I suspect that the new matrix class can more-or-less build off of `_MinimalMatrix`.

For those reasons, let's create a `_CastableMatrix` class with an `as_immutable()` and `as_mutable()` methods. Make the matrices in the tests that need it additionally inherit from `_CastableMatrix`